File: 프로그래머스/unrated/181928. 이어 붙인 수/이어 붙인 수.java
class Solution {
    public int solution(int[] num) {
        StringBuilder odd = new StringBuilder();
        StringBuilder even = new StringBuilder();
       for(int i : num)
           if(i % 2 == 0) even.append(i);
           else odd.append(i);
      
        return Integer.parseInt(odd.toString()) + Integer.parseInt(even.toString());
    }
}

File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> num_list) {
    string odd = "";
    string even = "";
    for(int i : num_list){
        i % 2 == 0 ? even += to_string(i) : odd += to_string(i);
    }
    return stoi(odd) + stoi(even);
}